### ISSUE TITLE
Implement REQUEST_DRIVING_DISTANCE_TO query percept 

### ISSUE THREAD

**Issue by dhixsingh** [agentsoz/bdi-abm-integration#17]
Implement query `queryPercept(agentId, "REQUEST_DRIVING_DISTANCE_TO", location)` that should return the driving  distance to the given `location`. 

Use case is when a BDI agent is re-planning and wants to choose between several evacuation destinations, perhaps wanting to choose the one closest to its current location.

 /cc @osbornejr

**Comment by osbornejr** [comment]
Hi @dhixsingh I am trying to look into this issue at the moment, the main hurdle I think is in the MatsimModel (~line 480).

 I have setup a REQUEST_DRIVING_DISTANCE case that would most likely have two location coordinates as input args-- one that was obtained using REQUEST_LOCATION, and the other coming from the list of evacuation point locations. 

But I will need some matsim mechanism to get a travel time estimate between the two locations. Do you know which that would be? I am looking at something like LeastCostPath but am unsure how to implement it. 

Or is this kind of thing on the MATSim side best left to Kai?  

Cheers

**Comment by dhixsingh** [comment]
@kainagel is there an easy way to query MATSim for the driving distance (not beeline) between two nodes on the network? I guess we will first want to find the route between the nodes given some routing mode (like freespeed), and then the distance (and maybe also estimated time) covered by that route.

**Comment by kainagel** [comment]
See 1df1786229ba3262541596812ce962765b963b59 and later in queryPercept.  Can't say if it will work; please try around a bit with it.  One issue is that you need to define the EvacRoutingMode; it will thus return the length associated with the routing mode, _not_ the shortest path by distance.  If you need the latter, you/we will need to set up a shortest path router.

**Comment by dhixsingh** [comment]
Thanks @kainagel. Requiring the `EvacRoutingMode` is fine. Will try and it out and let you know if we have more questions.

**Comment by dhixsingh** [comment]
Available since ee66066fea462412c78ad8a1d69d71ecf25b38c8. 

Usage example is here:
https://github.com/agentsoz/bdi-abm-integration/blob/ee66066fea462412c78ad8a1d69d71ecf25b38c8/examples/bushfire/src/main/java/io/github/agentsoz/ees/agents/Resident.java#L104-L106

Example output from `scenarios/mount-alexander-shire/campbells-creek-1/jill.out ` looks like this:
```
Time 00286 Resident 0    : received fire alert
Time 00286 Resident 0    : will start driving to shelter in (Castlemaine, coords=252145.230968,5894310.30443) at time 291.0
Time 00405 Resident 0    : is blocked at link 51823
Time 00405 Resident 0    : is currently between locations (321543051, coords=251458.994864993,5892667.574804958) and (60303155, coords=251444.56941373763,5892815.969128571)
Time 00405 Resident 0    : estimated driving distance to shelter accounting for detour is 3.9km
Time 00405 Resident 0    : will reevaluate route to destination (attempt 2 of 5)
Time 00405 Resident 0    : will start driving to shelter in (Castlemaine, coords=252145.230968,5894310.30443) at time 410.0
Time 00848 Resident 0    : arrived at shelter in (Castlemaine, coords=252145.230968,5894310.30443)
```
/cc @osbornejr
